src/components/AdjMatrix/AdjMatrix.vue

```diff
@@ -0,0 +1,216 @@
+<script lang='ts'>
+import * as d3 from 'd3';
+
+import { Model } from './model';
+import { View } from './view';
+import { Controller } from './controller';
+
+export default {
+  props: {
+    app: {
+      type: Object,
+      required: true,
+    },
+    provenance: {
+      type: Object,
+      required: true,
+    },
+    graphStructure: {
+      type: Object,
+      default: () => undefined,
+    },
+    selectNeighbors: {
+      type: Boolean,
+      default: true,
+    },
+  },
+
+  data() {
+    return {
+      browser: {
+        height: 0,
+        width: 0,
+      },
+      panelDimensions: { width: 0, height: 0 },
+      visDimensions: { width: 0, height: 0 },
+      visMargins: {
+        left: 25,
+        right: 25,
+        top: 25,
+        bottom: 25,
+      },
+      svg: undefined,
+      model: Model,
+      view: View,
+      controller: Controller,
+    };
+  },
+
+  computed: {
+    properties() {
+      const { graphStructure } = this;
+      return {
+        graphStructure,
+      };
+    },
+  },
+
+  watch: {
+    properties() {
+      //   this.updateVis();
+    },
+  },
+
+  async mounted() {
+    // Size the panel
+    this.browser.width = parseInt(
+      d3
+        .select('body')
+        .style('width')
+        .replace('px', ''),
+      0,
+    );
+
+    this.browser.height = parseInt(
+      d3
+        .select('body')
+        .style('height')
+        .replace('px', ''),
+      0,
+    );
+
+    // Set dimensions of the node link
+    this.visDimensions.width = this.browser.width * 0.75;
+    this.visDimensions.height = this.browser.height * 1;
+
+    // Set dimensions of panel
+    this.panelDimensions.width = this.browser.width * 0.25;
+    this.panelDimensions.height = this.browser.height * 1;
+
+    // Size the svg
+    this.svg = d3
+      .select(this.$refs.svg)
+      .attr('width', this.visDimensions.width)
+      .attr('height', this.visDimensions.height);
+
+    // Define the MVC
+    this.model = new Model(this.graphStructure);
+    this.view = new View();
+    this.controller = new Controller(
+      this.view,
+      this.model,
+      this.visDimensions,
+      this.panelDimensions,
+    );
+  },
+
+  methods: {},
+};
+</script>
+
+<template>
+  <div>
+    <svg ref="svg" width="800" height="900" />
+  </div>
+</template>
+
+<style>
```

Comment:
```suggestion
<style scoped>
```

You should probably use scoped styles for this; if this component ever gets embedded somewhere else, this could cause headache-inducing style bugs.

Reply:
This will require moving some of these rules up to controls.vue, etc. I think it's important we get this right now so I'll look into this change

Reply:
Might be better to do it in a followup PR, then. If everything works right now, then getting this merged and making incremental improvements is probably the way to go.

Reply:
Roger that. Opened #29